core/src/main/java/org/apache/brooklyn/core/location/LocationConfigKeys.java

```diff
@@ -35,7 +35,9 @@
     public static final ConfigKey<String> DISPLAY_NAME = ConfigKeys.newStringConfigKey("displayName");
     public static final ConfigKey<Boolean> ENABLED = ConfigKeys.newBooleanConfigKey("enabled", "Whether the location is enabled for listing and use "
         + "(only supported for selected locations)", true);
-    
+
+    public static final ConfigKey<String> EXTENSION = ConfigKeys.newStringConfigKey("extension");
```

Comment:
Better make it Map<String, Object> to
  * allow multiple extensions to be registered
  * use $brooklyn:object to construct the extensions (though brooklyn DSL doesn't work too well in location config). If it's a string then could treat it as class name and instantiate it. 

wdyt?